### PR TITLE
feat(billing): Update on-demand audit log to mention previous on-demand max spend.

### PR DESCRIPTION
**Pull request by dashed**
Depends on https://github.com/getsentry/getsentry/pull/6631

Screen Shot 2021-11-23 at 3 46 25 PM

### CODE REVIEW

File: src/sentry/models/auditlogentry.py [hunk 1]
```diff
@@ -361,7 +361,11 @@ def get_note(self):
         elif self.event == AuditLogEntryEvent.SET_ONDEMAND:
             if self.data["ondemand"] == -1:
                 return "changed on-demand spend to unlimited"
-            return "changed on-demand max spend to $%d" % (self.data["ondemand"] / 100,)
+            next_ondemand_max_spend = self.data["ondemand"] / 100
+            if "prev_ondemand" in self.data:
+                prev_ondemand_max_spend = self.data["prev_ondemand"] / 100
+                return f"changed on-demand max spend to ${next_ondemand_max_spend} from ${prev_ondemand_max_spend}"
```
Comment (maheskett) [inline src/sentry/models/auditlogentry.py:367]: Is there a function for displaying prices on the backend? If not, one could be added or something like `${next_ondemand_max_spend:.2f}` could be used for two decimal places. On the frontend, we generally show two decimal places only if there are cents in the price.

Comment (markstory) [reply src/sentry/models/auditlogentry.py:367]: > Is there a function for displaying prices on the backend?

I don't think there is one in sentry, as sentry doesn't have much in the way of pricing behavior. Formatting with `f"{val:.2f}` should be ok for an audit log though.

Comment (maheskett) [reply src/sentry/models/auditlogentry.py:367]: Yeah, it'd be great to extend the audit log to getsentry eventually. This is okay for now though.

**Comment by dashed** [comment]
@maheskett @markstory I've added a formatting function for on-demand max spends. Cents are display only if the on-demand max spends have non-zero cents.

This is what it looks like:
Screen Shot 2021-11-23 at 3 46 25 PM

File: src/sentry/models/auditlogentry.py [hunk 2]
```diff
@@ -361,7 +369,11 @@ def get_note(self):
         elif self.event == AuditLogEntryEvent.SET_ONDEMAND:
             if self.data["ondemand"] == -1:
                 return "changed on-demand spend to unlimited"
-            return "changed on-demand max spend to $%d" % (self.data["ondemand"] / 100,)
+            next_ondemand_max_spend = format_ondemand_max_spend(self.data["ondemand"])
+            if "prev_ondemand" in self.data:
+                prev_ondemand_max_spend = format_ondemand_max_spend(self.data["prev_ondemand"])
+                return f"changed on-demand max spend to {next_ondemand_max_spend} from {prev_ondemand_max_spend}"
```
Comment (maheskett) [inline src/sentry/models/auditlogentry.py:375]:
```suggestion
                return f"changed on-demand max spend from {prev_ondemand_max_spend} to {next_ondemand_max_spend}"
```

**Review by maheskett — APPROVED** [review]

**Pull request merged**